lib/ansible/plugins/connection/winrm.py

```diff
@@ -25,18 +25,23 @@
 import shlex
 import traceback
 import json
-import xmltodict
 
-from ansible.compat.six.moves.urllib.parse import urlunsplit
 
+from ansible.compat.six.moves.urllib.parse import urlunsplit
 from ansible.errors import AnsibleError, AnsibleConnectionFailure
+
 try:
     import winrm
     from winrm import Response
     from winrm.protocol import Protocol
 except ImportError:
     raise AnsibleError("winrm is not installed")
 
+try:
+    import xmltodict
+except ImportError:
+    raise AnsibleError("winrm is not installed")
+
```

Comment:
Now that I think about it, I definitely cargo cult'ed that import error idiom from the existing code. It doesn't seem to be common in the code base. 

There are a couple of places with similar requirements (try to import something absolutely required, but for an optional plugin...). Is there a preferred way to handle those cases? 

The try/except and AnsibleError seems reasonable (at least with the correct error message ;->). But there are a few approaches in the code:

plugins/cache/redis.py  also raises an AnsibleError
plugins/cache/memcached.py prints() an error and calls sys.exit()   (eek!)

[module_utils/basic.py does variations of sys.exit, but that's a special case...]